src/app/fdctl/run/tiles/fd_net.c

```diff
@@ -409,52 +460,173 @@ after_frag( void *             _ctx,
   *opt_filter = 1;
 }
 
+#define XSKMAP_FD (0x3FFFFFFF)
+
 static void
 privileged_init( fd_topo_t *      topo,
                  fd_topo_tile_t * tile,
                  void *           scratch ) {
-  (void)topo;
-
+  config_t * config = topo->config;  /* XXX ESCAPE HATCH REMOVE THIS */
+  
   FD_SCRATCH_ALLOC_INIT( l, scratch );
-  fd_net_init_ctx_t * ctx = FD_SCRATCH_ALLOC_APPEND( l, alignof( fd_net_init_ctx_t ), sizeof( fd_net_init_ctx_t ) );
-  FD_SCRATCH_ALLOC_APPEND( l, alignof( fd_net_ctx_t ),      sizeof( fd_net_ctx_t ) );
-  FD_SCRATCH_ALLOC_APPEND( l, fd_aio_align(),               fd_aio_footprint() );
-
-  /* Initialize XSK and xsk_aio which requires being privileged. */
-  void * xsk = fd_xsk_new( FD_SCRATCH_ALLOC_APPEND( l, fd_xsk_align(), fd_xsk_footprint( FD_NET_MTU, tile->net.xdp_rx_queue_size, tile->net.xdp_rx_queue_size, tile->net.xdp_tx_queue_size, tile->net.xdp_tx_queue_size ) ),
-                           FD_NET_MTU,
-                           tile->net.xdp_rx_queue_size,
-                           tile->net.xdp_rx_queue_size,
-                           tile->net.xdp_tx_queue_size,
-                           tile->net.xdp_tx_queue_size,
-                           tile->net.zero_copy );
-  if( FD_UNLIKELY( !fd_xsk_bind( xsk, tile->net.app_name, tile->net.interface, (uint)tile->kind_id ) ) )
-    FD_LOG_ERR(( "failed to bind xsk for net tile %lu", tile->kind_id ));
 
-  ctx->xsk = fd_xsk_join( xsk );
-  if( FD_UNLIKELY( !ctx->xsk ) ) FD_LOG_ERR(( "fd_xsk_join failed" ));
+  uint if_idx = if_nametoindex( tile->net.interface );
+  if( FD_UNLIKELY( !if_idx ) ) FD_LOG_ERR(( "if_nametoindex(%s) failed", tile->net.interface ));
+
+  uint lo_idx = if_nametoindex( "lo" );
+  if( FD_UNLIKELY( !lo_idx ) ) FD_LOG_ERR(( "if_nametoindex(lo) failed" ));
+
+  ulong net0_pid_obj_id = fd_pod_query_ulong( topo->props, "net0_pid", ULONG_MAX );
+  FD_TEST( net0_pid_obj_id!=ULONG_MAX );
+
+  int xsk_map_fd    = -1;
+  int lo_xsk_map_fd = -1;
+
+  if( tile->kind_id == 0 ) {
```

Comment:
Ugly to have all this logic in the same function. Split out and lightly comment for readability.